src/hooks/usePatchStatusSelect.ts

```diff
@@ -28,58 +83,96 @@ export const usePatchStatusSelect = (
-        }
+        newState[taskId] = nextCheckedState;
       });
     }
-
-    setSelectedTasks(newState);
+    dispatch({ type: "setSelectedTasks", data: newState });
   };
 
-  // Iterate through PatchBuildVariants and determine if a task should be selected or not
-  // Based on if the task status correlates with the validStatus filter
+  // Determine if a task is a selected based on the filter terms and available tasks
+  const prevPatchBuildVariants = usePrevious(patchBuildVariants);
+  const prevPatchStatusFilterTerm = usePrevious(patchStatusFilterTerm);
+  const prevBaseStatusFilterTerm = usePrevious(baseStatusFilterTerm);
   useEffect(() => {
-    if (patchBuildVariants) {
-      let tempSelectedTasks = selectedTasks;
-      patchBuildVariants.forEach((patchBuildVariant) => {
-        patchBuildVariant.tasks.forEach((task) => {
-          if (validStatus.includes(task.status)) {
-            tempSelectedTasks = addTaskToSelectedTasks(
-              task.id,
-              tempSelectedTasks
-            );
-          } else {
-            tempSelectedTasks = removeTaskFromSelectedTasks(
-              task.id,
-              tempSelectedTasks
-            );
-          }
+    const filterTermOrPatchTasksChanged =
+      patchBuildVariants !== prevPatchBuildVariants ||
+      patchStatusFilterTerm !== prevPatchStatusFilterTerm ||
+      baseStatusFilterTerm !== prevBaseStatusFilterTerm;
+    if (filterTermOrPatchTasksChanged) {
+      if (window.Worker && webWorker) {
+        webWorker.postMessage({
+          patchBuildVariants,
+          patchStatusFilterTerm,
+          baseStatusFilterTerm,
+          selectedTasks,
         });
-      });
-      setSelectedTasks(tempSelectedTasks);
+      } else {
+        // fallback in case web workers are not available.
+        // This code reflects logic in public/web_worker/patchBuildVariantReduce.js
+        // Iterate through PatchBuildVariants and determine if a task should be
+        // selected or not based on if the task status correlates with the 2 filters.
+        // if 1 of the 2 filters is empty, ignore the empty filter
+        const baseStatuses = new Set(baseStatusFilterTerm);
+        const statuses = new Set(patchStatusFilterTerm);
+        const nextState =
+          patchBuildVariants?.reduce(
+            (accumA, patchBuildVariant) =>
+              patchBuildVariant.tasks?.reduce(
+                (accumB, task) => ({
+                  ...accumB,
+                  [task.id]:
+                    (!!patchStatusFilterTerm?.length ||
+                      !!baseStatusFilterTerm?.length) &&
+                    (patchStatusFilterTerm?.length
+                      ? statuses.has(task.status)
+                      : true) &&
+                    (baseStatusFilterTerm?.length
+                      ? baseStatuses.has(task.baseStatus)
+                      : true),
+                }),
+                accumA
+              ),
+            { ...selectedTasks }
```

Comment:
Having this repeated code is why i made the suggestion above we should minimize this if at all possible.